File: spring-web-java/src/main/java/spring/web/java/presentation/exception/DuplicateEntityException.java
package spring.web.java.presentation.exception;

import org.springframework.http.HttpStatus;

public class DuplicateEntityException extends BusinessException {

	public DuplicateEntityException(Class<?> clazz, String name) {
		super("이미 존재하는 %s 엔티티입니다. (ID: '%s')".formatted(clazz.getSimpleName(), name), HttpStatus.CONFLICT);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/exception/DuplicateEntityException.kt
package spring.web.kotlin.presentation.exception

import org.springframework.http.HttpStatus

class DuplicateEntityException(clazz: Class<*>, name: String) :
    AbstractHttpException("이미 존재하는 ${clazz.simpleName} 엔티티입니다. (ID: '$name')", HttpStatus.CONFLICT)